src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,80 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { FC, useContext, useCallback } from 'react';
+import { UserContext } from '../UserContext';
+
+function debounce(f:(finalQuery: string) => void, delay: number) {
+  let timerId: NodeJS.Timeout;
+
+  return (arg: string) => {
```

Comment:
will be great to handle all available arguments, that can be passed to the debounce function, for now you can't reuse it for the function with other argument

Reply:
don't know why but only this option work